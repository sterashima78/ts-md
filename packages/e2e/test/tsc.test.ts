import { execSync } from 'node:child_process';
import fs from 'node:fs/promises';
import path from 'node:path';
import { describe, expect, it } from 'vitest';

const fixture = path.join(__dirname, 'fixtures', 'tsc');
const tsconfig = path.join(fixture, 'tsconfig.json');

async function runTsc() {
  execSync(`pnpm exec ts-md-tsc -p ${tsconfig} --emitDeclarationOnly`, {
    cwd: fixture,
    stdio: 'inherit',
  });
}

describe('ts-md-tsc', () => {
  it('generates dts for mixed refs', async () => {
    await runTsc();
    const out = await fs.readFile(
      path.join(fixture, 'dist', 'dep.ts.md.d.ts'),
      'utf8',
    );
    expect(out).toMatchSnapshot();
  });
});
