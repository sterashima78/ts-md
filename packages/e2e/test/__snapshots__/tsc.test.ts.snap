// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ts-md-tsc > generates dts for mixed refs 1`] = `
"declare function bar_bar(): string;
export { bar_bar as bar };
"
`;
